torch/fx/experimental/proxy_tensor.py

```diff
@@ -1115,20 +1115,19 @@ def get_innermost_proxy_mode():
 
 
 @contextlib.contextmanager
-def disable_proxy_modes_tracing(enable_current=False):
-    # enable_current=True is now a no-op, since only one proxy mode
-    # can live on the stack at a time.
-    # We should kill this API in a future PR.
-    maybe_old = None
-    if not enable_current:
-        # Only one proxy_mode can be "active" at a time.
-        # So we simply remove our active mode.
-        maybe_old = torch._C._unset_dispatch_mode(torch._C._TorchDispatchModeKey.PROXY)
+def disable_proxy_modes_tracing(pre_dispatch=False):
```

Comment:
Hmm.. remind me if this is wrong, but I would think that:

(1) a `pre_dispatch` kwarg to this function is not necessary

(2) instead, you can query both the normal mode stack and the pre-dispatch mode stack, and assert that exactly one of them has a proxy mode on it.

(3) When you pop that mode, it should already have a `_dispatch_key` piece of state on it telling us whether or not it was a pre-dispatch mode or a normal mode, so you can use that to know which stack to push it onto after the yield.

Reply:
In fact - this is exactly what you're doing for `unset_functional_temporarily()` :) Seems reasonable to give this fn the same treatment